helm-v3/005-test.md

```diff
@@ -0,0 +1,27 @@
+# Test Command
+
+## Overview
+
+`helm test` gives a chart author the ability to define actions on a helm release which validate that the chart is running as intended. This is useful for ensuring that a `helm install` or `helm upgrade` created/modified Kubernetes resources successfully and that the consequential `helm release` is functional.
+* A **test** is defined as a command and some arguments executed from a container running in a Kubernetes Pod. If the command completes inside the container with an exit code of 0, it is successful. All other exit codes indicate the test failed.
+* Tests are defined in Manifests that live inside of the `templates/tests/` directory inside of a chart.
+* Tests use annotations to denote what the expected condition is. 
+    * If a test is expected to succeed, it will have the following annotation: `helm.sh/test-expect-success: true`
+    * If a test is expected to fail, it will have the following annotation: `helm.sh/test-expect-success: false`
+* `helm test` commands
```

Comment:
Could we have a way to define groups of tests (suites)?

For example on first install I'd like to be able to run a "full-verification" suite, then on each upgrade run a "verify-upgrade" suite.

Reply:
oh yeah that's a good idea. I think we can add an annotation to make that possible.

Reply:
@viglesiasce do you think we should group tests via an annotation or a label or using the directory structure?

Reply:
I think annotations are better option

Reply:
Annotations sound great.

Reply:
I would also prefer to use either a directory structure convention OR annotation for specifying test resources. Folks are used to annotations already, and annotations also provide the flexibility to define tests in whatever directory structure seems best for the chart, so I'm in favor of only using annotations.